# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1495987368 14400
#      Sun May 28 12:02:48 2017 -0400
# Node ID 94968aa621a8cfb9724abf1f43db8db566ffa6d2
# Parent  2aab625b502caf3c52d5e41a767223b605b11ee3
show stack trace for errors in command line and startup files (bug #49346)

* interpreter.cc (safe_source_file): Display stack trace for execution
errors.

diff --git a/libinterp/corefcn/interpreter.cc b/libinterp/corefcn/interpreter.cc
--- a/libinterp/corefcn/interpreter.cc
+++ b/libinterp/corefcn/interpreter.cc
@@ -323,18 +323,23 @@ safe_source_file (const std::string& fil
       octave::source_file (file_name, context, verbose, require_file, warn_for);
     }
   catch (const octave::interrupt_exception&)
     {
       octave::interpreter::recover_from_exception ();
 
       return 1;
     }
-  catch (const octave::execution_exception&)
+  catch (const octave::execution_exception& e)
     {
+      std::string stack_trace = e.info ();
+
+      if (! stack_trace.empty ())
+        std::cerr << stack_trace;
+
       octave::interpreter::recover_from_exception ();
 
       return 1;
     }
 
   return 0;
 }
 
